src/model/Course.java

```diff
@@ -12,36 +12,949 @@
  * 
  */
 public class Course {
-	
-	
+
+	// Field(s)
+
 	/**
 	 * Complete name of the course
 	 */
-	String courseName;
-	
+	private String courseName;
+
 	/**
-	 * Unique identifier for a course. 
+	 * Unique identifier for a course
 	 */
-	String code;
-	
+	private String code;
+
 	/**
-	 * Creation date of the course syllabus to differentiate
-	 * this syllabus from others for the same course
+	 * Creation date of the course syllabus to differentiate this syllabus from
+	 * others for the same course
 	 */
-	Date creationDate;
-	
+	private LocalDateTime creationDate;
+
 	/**
 	 * The courses that should be completed before taking this one
 	 */
-	ArrayList<String> prerequisites;
-	
+	private ArrayList<String> prerequisites;
+
 	/**
 	 * Specifies the course language
 	 */
-	Language courseLang;
+	private Language courseLang;
+
+	/**
+	 * Specifies the semester(s) the course will be given
+	 */
+	private SemesterOptions semester;
+
+	/**
+	 * Shows the local credit for the course and calculated by the program itself
+	 * according to the pseudo code below ; localCredit = round(theoreticalHour +
+	 * (0.5 * labHour))
+	 * 
+	 * @see #calculateLocalCredit()
+	 */
+	private int localCredit;
+
+	/**
+	 * Shows the ECTS equivalent value of the course This value is calculated by the
+	 * program itself according the pseudo code below ; ECTS = round(sumAll(workload
+	 * of each element in workloadTable) / 30)
+	 * 
+	 * @see #calculateECTS()
+	 *
+	 */
+	private int ects;
+
+	/**
+	 * Theoretical hours per week for the course.
+	 */
+	private int theoreticalHour;
+
+	/**
+	 * Lab/Application hours per week for the course.
+	 */
+	private int labHour;
+
+	/**
+	 * Course type
+	 */
+	private CourseType type;
+
+	/**
+	 * Coordinator of the course. There can only be 1 coordinator and he must be a
+	 * teacher
+	 */
+	private Lecturer courseCordinator;
+
+	/**
+	 * Lecturers who give this course. There can be more then one lecturer
+	 * 
+	 * @see #addLecturer(Lecturer)
+	 * @see #removeLecturer(Lecturer)
+	 */
+	private ArrayList<Lecturer> courseLecturers;
```

Comment:
merak ettiğim bir nokta var burada, internetten gelen syllabuslar için hocaları nasıl yapmamız gerekiyor? mesela iki adet hoca geliyor burada, Turhan hoca ve Cem hoca, onları geldiğinde tanımalı mı sistem?

Reply:
aynı konu asistanlar için de geçerli

Reply:
Eklemek için gerekli tüm bilgiler zaten ders kordinatörleri, hocalar veya asistanlar için müfredat programında yer alıyor. (Title) . (Name) (Surname) olacak şekilde. Bu 3 veriyi düzenli ifadeler (regex) ile ayırıp sonrasında programa ulaştıktan sonra Lecturer class'ının bir örneği olarak yaratılacak ve Course class'ına ait ilk constructor'a (dökümentasyon bölümünde bu amaçla kullanılacağı belirtilen constructor) aktarılacak.